docs/hub/models-adding-libraries.md

```diff
@@ -145,51 +120,69 @@ For example:
 
-       ```python
-       ALLOWED_TASKS: Dict[str, Type[Pipeline]] = {
-           "token-classification": TokenClassificationPipeline
-       }
-       ```
+To register a new library, please open a Pull Request [here](https://github.com/huggingface/huggingface.js/blob/main/packages/tasks/src/model-libraries.ts) following the instructions below:
+- The library id should be lowercased and hyphen-separated (example: `"adapter-transformers"`). Make sure to preserve alphabetical order when opening the PR.
+- set `repoName` and `prettyLabel` with user-friendly casing (example: `DeepForest`).
+- set `repoUrl` with a link to the library source code (usually a GitHub repository).
+- (optional) set `docsUrl` with a link to the docs of the library. If the documentation is in the GitHub repo referenced above, no need to set it twice.
+- set `filter` to `false`.
+- (optional) define how downloads must be counted by setting `countDownload`. Downloads can be tracked by file extensions or filenames. Make sure to not duplicate the counting. For instance, if loading a model requires 3 files, the download count rule must count downloads only on 1 of the 3 files. Otherwise, the download count will be overestimated.
+**Note:** if the library uses one of the default config files (`config.json`, `config.yaml`, `hyperparams.yaml`, and `meta.yaml`, see [here](https://huggingface.co/docs/hub/models-download-stats#which-are-the-query-files-for-different-libraries)), there is no need to manually define a download count rule.
+- (optional) define `snippets` to let the user know how they can quickly instantiate a model. More details below.
 
-    * For each task your library supports, modify the `app/pipelines/task_name.py` files accordingly. We have also added an `IMPLEMENT_THIS` flag in the pipeline files to guide you. If there isn't a pipeline that supports your task, feel free to add one. Open an [issue](https://github.com/huggingface/huggingface.js/issues/new) here, and we will be happy to help you.
-    * Add your model and task to the `tests/test_api.py` file. For example, if you have a text generation model:
+Before opening the PR, make sure that at least one model is referenced on https://huggingface.co/models?other=my-library-name. If not, the model card metadata of the relevant models must be updated with `library_name: my-library-name` (see [example](https://huggingface.co/google/gemma-scope/blob/main/README.md?code=true#L3)). If you are not the owner of the models on the Hub, please open PRs (see [example](https://huggingface.co/MCG-NJU/VFIMamba/discussions/1)).
 
-       ```python
-       TESTABLE_MODELS: Dict[str,str] = {
-           "text-generation": "my-gpt2-model"
-       }
-       ```
-3. Finally, run the following test to ensure everything works as expected:
+Here is a minimal [example](https://github.com/huggingface/huggingface.js/pull/561/files) adding integration for Grok-1.
```

Comment:
Should we have an example including a snippet and a simpler download rule?

Reply:
What about this one? https://github.com/huggingface/huggingface.js/pull/885/files - It looks quite clean and has download counts and snippets both.

Reply:
Created a suggestion.

Reply:
Thanks! The VFI Mamba PR is indeed very nice with a complete PR and still concise. I've merged your suggestiong @Vaibhavs10 + updated the second link (Grok-1 PR was referenced twice). We should be good now!